fuse_models/include/fuse_models/common/sensor_proc.h

```diff
@@ -465,7 +461,197 @@ inline bool processDifferentialPoseWithCovariance(
     pose2.pose.covariance[31],
     pose2.pose.covariance[35];
 
-  auto pose_relative_covariance = j_pose1 * cov1 * j_pose1.transpose() + j_pose2 * cov2 * j_pose2.transpose();
+  fuse_core::Matrix3d pose_relative_covariance;
+  if (independent)
+  {
+    // Compute Jacobians so we can rotate the covariance
+    fuse_core::Matrix3d j_pose1;
+    j_pose1 <<
+      -cy,  sy,  sy * x_diff + cy * y_diff,
+      -sy, -cy, -cy * x_diff + sy * y_diff,
+        0,   0,                         -1;
+
+    fuse_core::Matrix3d j_pose2;
+    j_pose2 <<
+       cy, -sy,  0,
+       sy,  cy,  0,
+        0,   0,  1;
+
+    pose_relative_covariance = j_pose1 * cov1 * j_pose1.transpose() + j_pose2 * cov2 * j_pose2.transpose();
```

Comment:
I also try this naive change for the `independent == false` case, using the jacobians from this `if (independent)` block: `pose_relative_covariance = j_pose2 * cov2 * j_pose2.transpose() - j_pose1 * cov1 * j_pose1.transpose();`

And also adding a `minimum_pose_relative_covariance` to the resulting `pose_relative_covariance` of that, to avoid zero covariance matrices when `cov1 == cov2`.